### PR TITLE
Alan Van Test

### PR DESCRIPTION
**Pull request by alanvan0502**
Submission of my test please

### CODE REVIEW

**Comment by markhollands** [comment]
Hey @alanvan0502 , thanks for submitting the PR.  Does your code work with Retrofit now?  It still seems to be using the AsyncTask to get the data.  If you want to spend some more time to remove the legacy code (AsyncTask/Json Parser), you're welcome to spend a bit more time if you want.

**Comment by alanvan0502** [comment]
Hi Mark. I have just seen your comment. I will spend a bit of time to clean up the legacy code right now. Thank you!

File: app/src/main/java/com/lifx/lifxtest/MainActivity.kt [hunk 1]
```diff
@@ -1,41 +1,74 @@
...
 import org.json.JSONObject
 
 
 class MainActivity : AppCompatActivity() {
 
     lateinit var adapter: ListAdapter
-    val listData = mutableListOf<String>()
+    val listData = mutableListOf<Data>()
+    private lateinit var viewModel: AppViewModel
+    private var bag: CompositeDisposable? = null
```
Comment (markhollands) [inline app/src/main/java/com/lifx/lifxtest/MainActivity.kt]: @alanvan0502  Just curious why you chose to make your CompositeDisposable nullable?

Comment (alanvan0502) [reply app/src/main/java/com/lifx/lifxtest/MainActivity.kt]: @markhollands it's just one of the ways that I declare it. However, if I were to refactor the code, I would likely change it it to: `private val bag = CompositeDisposable()`

Comment (alanvan0502) [reply app/src/main/java/com/lifx/lifxtest/MainActivity.kt]: @markhollands Furthermore, I would change the access modifier of _listData_ to `private val listData` and of _adapter_ to `private lateinit var adapter: ListAdapter`

**Comment by markhollands** [comment]
@alanvan0502 Thanks for finishing off your test and discussing the review.  We be running through some more interviews over this week / early next week, but we'll let you know how your application has gone next week.

**Pull request closed without merging**